docs/serving/model-api.md

```diff
@@ -0,0 +1,239 @@
+
+Model serving classes must inherit from `mlrun.serving.V2ModelServer`, and at the minimum 
+implement the `load()` (download the model file(s) and load the model into memory) 
+and `predict()` (accept request payload and return prediction/inference results) methods.  
+
+The class is initialized automatically by the model server and can run locally
+as part of a nuclio serverless function, or as part of a real-time pipeline
+
+You need to implement two mandatory methods:
+  * **load()**     - download the model file(s) and load the model into memory, 
+  note this can be done synchronously or asynchronously 
+  * **predict()**  - accept request payload and return prediction/inference results
+
+you can override additional methods : `preprocess`, `validate`, `postprocess`, `explain`<br>
+you can add custom api endpoint by adding method `op_xx(event)`, it can be invoked by
+calling the <model-url>/xx (operation = xx)
+
+**minimal sklearn serving function example:**
+
+```python
+from cloudpickle import load
+import numpy as np
+import mlrun
+
+class ClassifierModel(mlrun.serving.V2ModelServer):
+    def load(self):
+        """load and initialize the model and/or other elements"""
+        model_file, extra_data = self.get_model('.pkl')
+        self.model = load(open(model_file, 'rb'))
+
+    def predict(self, body: dict) -> list:
+        """Generate model predictions from sample"""
+        feats = np.asarray(body['inputs'])
+        result: np.ndarray = self.model.predict(feats)
+        return result.tolist()
+```
+
+
+### Load() method
+
+in the load method we download the model from external store, run the algorithm/framework
+`load()` call, and do any other initialization logic. 
+
+load will run synchronously (the deploy will be stalled until load completes), 
+this can be an issue for large models and cause a readiness timeout, we can increase the 
+function `spec.readiness_timeout`, or alternatively choose async loading (load () will 
+run in the background) sy setting the function `spec.load_mode = "async"`.  
+
+the function `self.get_model()` downloads the model metadata object and main file (into `model_file` path),
+additional files can be accessed using the returned `extra_data` (dict of dataitem objects).
+
+the model metadata object is stored in `self.model_spec` and provide model parameters, metrics, schema, etc.
+parameters can be accessed using `self.get_param(key)`, the parameters can be specified in the model or during 
+the function/model deployment.  
+
+### predict() method
+
+the predict method is called when we access the `/infer` or `/predict` url suffix (operation).
+the method accepts the request object (as dict), see [API doc](#model-server-api) below.
+and should return the specified response object.
+
+### explain() method
+
+the explain method provides a hook for model explanability, and is accessed using the `/explain` operation. .
```

Comment:
Do we have an explain example? I don't think I've seen one

Reply:
nope, it require a bunch of work